xarray/tests/test_combine.py

```diff
@@ -229,8 +228,22 @@ def test_lexicographic_sort_string_coords(self):
         assert concat_dims == ["simulation"]
 
     def test_datetime_coords(self):
-        ds0 = Dataset({"time": [datetime(2000, 3, 6), datetime(2001, 3, 7)]})
-        ds1 = Dataset({"time": [datetime(1999, 1, 1), datetime(1999, 2, 4)]})
+        ds0 = Dataset(
+            {
+                "time": [
+                    np.datetime64("2000-03-06", "ns"),
+                    np.datetime64("2000-03-07", "ns"),
+                ]
+            }
+        )
+        ds1 = Dataset(
+            {
+                "time": [
+                    np.datetime64("1999-01-01", "ns"),
+                    np.datetime64("1999-02-04", "ns"),
+                ]
+            }
+        )
```

Comment:
Construct these Datasets with `np.datetime64[ns]` values directly instead of `datetime.datetime` objects, which lead to a warning under pandas 3.

Reply:
is there a reason why we use a list of datetimes instead of a datetime array?
```python
np.array(["1999-01-01", "1999-02-04"], dtype="datetime64[ns]")
```

Reply:
No, indeed that's better—thanks.